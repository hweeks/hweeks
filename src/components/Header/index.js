import React from 'react';
import { Link } from 'react-router-dom';
import {
  HeaderContainer,
  HeaderLinkWrapped,
  HeaderLogoWrapped,
  HeaderLink,
  HeaderButton,
  HeaderMenuWrapper,
} from './styles'

export default class Header extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = { isVisible: false };
    this.toggleMenu = this.toggleMenu.bind(this);
    this.closeMenu = this.closeMenu.bind(this);
  }
  getToggleState() {
    let menuClass = 'header__links';
    if ((this.state || {}).isVisible) {
      menuClass = `${menuClass} active`;
    }
    return menuClass;
  }
  closeMenu() {
    this.setState({
      isVisible: false,
    });
  }
  toggleMenu() {
    this.setState({
      isVisible: !this.state.isVisible,
    });
  }
  render() {
    const {isVisible} = this.state
    return (
      <HeaderContainer>
        <HeaderLogoWrapped to="/" onClick={this.closeMenu}>
          hweeks.com
        </HeaderLogoWrapped>
        <HeaderButton role="menuitem" onClick={this.toggleMenu}>
          menu
        </HeaderButton>
        <HeaderMenuWrapper isVisible={isVisible}>
          <HeaderLinkWrapped to="/bikes" onClick={this.toggleMenu}>
            bikes
          </HeaderLinkWrapped>
          <HeaderLinkWrapped to="/blog" onClick={this.toggleMenu}>
            blog
          </HeaderLinkWrapped>
          <HeaderLink href="/assets/pdf/r.pdf" role="menuitem" onClick={this.toggleMenu}>
            resume
          </HeaderLink>
          <HeaderLink href="https://github.com/hweeks">git</HeaderLink>
          <HeaderLink
            href="mailto:root@hweeks.com"
            role="menuitem"
            onClick={this.toggleMenu}
          >
            email
          </HeaderLink>
        </HeaderMenuWrapper>
      </HeaderContainer>
    );
  }
}
