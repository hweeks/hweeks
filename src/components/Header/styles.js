import styled from 'styled-components'
import { Link } from 'react-router-dom'
import baseStyles from '../../styleRoot'
const {atom, borderDefault, bgColor, headerFontSize, breakpoint} = baseStyles

export const HeaderContainer = styled.nav`
  display: flex;
  flex-flow: row;
  justify-content: space-between;
  align-items: center;
  position: relative;
  padding: ${atom * 2}rem ${atom}rem;
  border-bottom: ${borderDefault};
`

export const HeaderLinkWrapped = styled(Link)`
  line-height: ${headerFontSize}rem;
  font-size: ${headerFontSize}rem;
  padding: ${atom * 2}rem;
  text-decoration: none;
  color: inherit;
`

export const HeaderLogoWrapped = styled(Link)`
  font-size: ${headerFontSize}rem;
  text-decoration: none;
  color: inherit;
  @media (min-width: ${breakpoint}rem) {
    padding: 0 ${atom * 2}rem;
  }
`

export const HeaderLink = styled.a`
  line-height: ${headerFontSize}rem;
  font-size: ${headerFontSize}rem;
  padding: ${atom * 2}rem;
  text-decoration: none;
  color: inherit;
  @media (min-width: ${breakpoint}rem) {
    padding: 0 ${atom * 2}rem;
  }
`

export const HeaderButton = styled.button`
  font-size: ${headerFontSize}rem;
  border: 0;
  margin: 0;
  padding: 0;
  background-color: inherit;
  outline: 0;
  @media (min-width: ${breakpoint}rem) {
    display: none;
  }
`

export const HeaderMenuWrapper = styled.div`
  display: ${props => props.isVisible ? 'flex' : 'none'};
  position: absolute;
  flex-flow: column;
  justify-content: center;
  align-items: center;
  border-bottom: ${borderDefault};
  width: 100%;
  bottom: -${atom / 2}rem;
  left: 0;
  transform: translateY(100%);
  background: ${bgColor};
  @media (min-width: ${breakpoint}rem) {
    display: flex;
    flex-flow: row;
    position: static;
    border: 0;
    transform: none;
    width: auto;
  }
`
