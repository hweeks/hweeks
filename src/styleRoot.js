const rootValues = {
  atom: '0.4',
  headerFontSize: '2.2',
  bodyFontSize: '1.6',
  borderColor: '#666',
  bgColor: '#fff',
  borderDefault: `0.2rem solid #666`,
  breakpoint: '41.5',
}

export default rootValues
