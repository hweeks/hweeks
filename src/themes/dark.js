export const dark = {
  common: {
    accent: '#E6B450',
    bg: '#0A0E14',
    fg: '#B3B1AD',
    ui: '#3D424D',
  },
  syntax: {
    tag: '#39BAE6',
    func: '#FFB454',
    entity: '#59C2FF',
    string: '#C2D94C',
    regexp: '#95E6CB',
    markup: '#F07178',
    keyword: '#FF8F40',
    special: '#E6B673',
    comment: '#626A73',
    constant: '#FFEE99',
    operator: '#F29668',
    error: '#FF3333',
  },
  ui: {
    line: '#01060E',
    panel: {
      bg: '#0D1016',
      shadow: '#00010A',
      border: '#000000',
    },
    gutter: {
      normal: '#3D424D99',
      active: '#3D424DE6',
    },
    selection: {
      bg: '#273747',
      inactive: '#1F2C39',
      border: '#304357',
    },
    guide: {
      active: '#3D424DB3',
      normal: '#3D424D66',
    },
  },
  vcs: {
    added: '#91B362',
    modified: '#6994BF',
    removed: '#D96C75',
  },
};
