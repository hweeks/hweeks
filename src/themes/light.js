export const light = {
  common: {
    accent: '#FF9940',
    bg: '#FAFAFA',
    fg: '#6C7680',
    ui: '#959DA6',
  },
  syntax: {
    tag: '#55B4D4',
    func: '#F2AE49',
    entity: '#399EE6',
    string: '#86B300',
    regexp: '#4CBF99',
    markup: '#F07171',
    keyword: '#FA8D3E',
    special: '#E6BA7E',
    comment: '#ABB0B6',
    constant: '#A37ACC',
    operator: '#ED9366',
    error: '#F51818',
  },
  ui: {
    line: '#959DA61A',
    panel: {
      bg: '#FFFFFF',
      shadow: '#56606940',
      border: '#F0F0F0',
    },
    gutter: {
      normal: '#959DA666',
      active: '#959DA6CC',
    },
    selection: {
      bg: '#EDF0F5',
      inactive: '#F2F4F7',
      border: '#E5EBF2',
    },
    guide: {
      active: '#959DA6B3',
      normal: '#959DA64D',
    },
  },
  vcs: {
    added: '#99BF4D',
    modified: '#709ECC',
    removed: '#F27983',
  },
};
