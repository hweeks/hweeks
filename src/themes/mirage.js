export const mirage = {
  common: {
    accent: '#FFCC66',
    bg: '#1F2430',
    fg: '#CBCCC6',
    ui: '#707A8C',
  },
  syntax: {
    tag: '#5CCFE6',
    func: '#FFD580',
    entity: '#73D0FF',
    string: '#BAE67E',
    regexp: '#95E6CB',
    markup: '#F28779',
    keyword: '#FFA759',
    special: '#FFE6B3',
    comment: '#5C6773',
    constant: '#D4BFFF',
    operator: '#F29E74',
    error: '#FF3333',
  },
  ui: {
    line: '#191E2A',
    panel: {
      bg: '#232834',
      shadow: '#141925',
      border: '#101521',
    },
    gutter: {
      normal: '#707A8C66',
      active: '#707A8CCC',
    },
    selection: {
      bg: '#34455A',
      inactive: '#2D3B4D',
      border: '#3C526A',
    },
    guide: {
      active: '#707A8CB3',
      normal: '#707A8C4D',
    },
  },
  vcs: {
    added: '#A6CC70',
    modified: '#77A8D9',
    removed: '#F27983',
  },
};
