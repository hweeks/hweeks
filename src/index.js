import ReactDOM from 'react-dom';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import styled, { ThemeProvider } from 'styled-components';
import App from './composite/App';
import * as theme from './themes';

const ThemeSelect = styled.div`
  display: ${props => (props.display ? 'block' : 'none')};
`;

const ThemeSelectWrapper = styled.div`
  position: fixed;
  bottom: 0;
`;

class AppWrapper extends React.PureComponent {
  constructor(props) {
    super(props);
    const themes = ['dark', 'light', 'mirage'];
    this.handleThemeSelect = this.handleThemeSelect.bind(this);
    this.toggleThemer = this.toggleThemer.bind(this);
    this.state = {
      theme: theme[themes[Math.floor(Math.random() * 3) % 3]],
      menuShown: false,
    };
  }
  handleThemeSelect({ target }) {
    if (theme[target.value]) {
      this.setState({
        theme: theme[target.value],
        menuShown: false,
      });
    }
  }
  toggleThemer() {
    this.setState({
      menuShown: !this.state.menuShown,
    });
  }
  render() {
    return (
      <BrowserRouter>
        <ThemeProvider theme={this.state.theme}>
          <React.Fragment>
            <App />
            <ThemeSelectWrapper>
              <button onClick={this.toggleThemer}>
                theme
              </button>
              <ThemeSelect display={this.state.menuShown}>
                <button onClick={this.handleThemeSelect} value="light">light</button>
                <button onClick={this.handleThemeSelect} value="dark">dark</button>
                <button onClick={this.handleThemeSelect} value="mirage">mirage</button>
              </ThemeSelect>
            </ThemeSelectWrapper>
          </React.Fragment>
        </ThemeProvider>
      </BrowserRouter>
    );
  }
}

const load = () => {
  ReactDOM.render(
    <AppWrapper />,
    document.querySelector('#home'),
  );
};

if (document.readyState !== 'complete') {
  document.addEventListener('DOMContentLoaded', load);
} else {
  load();
}
