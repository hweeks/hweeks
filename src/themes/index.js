import { light as lTheme } from './light';
import { mirage as mTheme } from './mirage';
import { dark as dTheme } from './dark';

const rootValues = {
  atom: '0.4',
  headerFontSize: '2.2',
  bodyFontSize: '1.6',
  borderColor: '#666',
  bgColor: '#fff',
  borderDefault: '0.2rem solid #666',
  breakpoint: '41.5',
};

export const light = { ...rootValues, ...lTheme };

export const dark = { ...rootValues, ...dTheme };

export const mirage = { ...rootValues, ...mTheme };
