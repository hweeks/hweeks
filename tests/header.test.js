import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';
import Header from '../src/components/Header';
import { ThemeProvider } from 'styled-components'
import { light } from '../src/themes'

const jestComponent = renderer.create(
  <ThemeProvider theme={light} >
    <StaticRouter context={{}}>
      <Header />
    </StaticRouter>
  </ThemeProvider>,
);

test('Header actually renders', () => {
  const tree = jestComponent.toJSON();
  expect(tree).toMatchSnapshot();
});
