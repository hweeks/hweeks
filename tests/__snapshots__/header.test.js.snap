// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header actually renders 1`] = `
<nav
  className="sc-bdVaJa cupZnc"
>
  <a
    className="sc-htpNat eVwUtI"
    href="/"
    onClick={[Function]}
  >
    hweeks.com
  </a>
  <button
    className="sc-ifAKCX dBisCF"
    onClick={[Function]}
    role="menuitem"
  >
    menu
  </button>
  <div
    className="sc-EHOje rSFJP"
  >
    <a
      className="sc-bwzfXH igPBjT"
      href="/bikes"
      onClick={[Function]}
    >
      bikes
    </a>
    <a
      className="sc-bwzfXH igPBjT"
      href="/blog"
      onClick={[Function]}
    >
      blog
    </a>
    <a
      className="sc-bxivhb gSSLLD"
      href="/assets/pdf/r.pdf"
      onClick={[Function]}
      role="menuitem"
    >
      resume
    </a>
    <a
      className="sc-bxivhb gSSLLD"
      href="https://github.com/hweeks"
    >
      git
    </a>
    <a
      className="sc-bxivhb gSSLLD"
      href="mailto:root@hweeks.com"
      onClick={[Function]}
      role="menuitem"
    >
      email
    </a>
  </div>
</nav>
`;
